### PR TITLE
Feat/files schema validation

### PR DESCRIPTION
**Pull request by smelm**
Replaces fluent-ts-validator for most file requests. This is achieved with a combination of [ts-json-schema-generator](https://github.com/vega/ts-json-schema-generator) and [ajv](https://ajv.js.org/).

The most significant additions are `SchemaRepository` and `SchemaValidator`.

During startup there is a performance hit that is caused by `ts-json-schema-generator` which does not respect the path to the request files and instead parses the entire project. If there is no easy fix for this it can be avoided be looping through the requests ourselves. Edit: this does not matter anymore, since it happens at build time.

### CODE REVIEW

File: src/useCases/common/SchemaRepository.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+const customFormats: string[][] = [
+    ["bkb-file", "FIL[A-Za-z0-9]{17}"],
+    ["bkb-relationship", "REL[A-Za-z0-9]{17}"],
+    ["bkb-relationshipRequest", "RRQ[A-Za-z0-9]{17}"],
+    ["bkb-message", "MSG[A-Za-z0-9]{17}"],
+    ["bkb-relationshipTemplate", "RLT[A-Za-z0-9]{17}"],
+    ["bkb-token", "TOK[A-Za-z0-9]{17}"],
+    ["bkb-relationshipChange", "RCH[A-Za-z0-9]{17}"],
+    ["bkb-device", "DVC[A-Za-z0-9]{17}"]
+];
```
Comment (smelm) [inline src/useCases/common/SchemaRepository.ts:24]: These should probably be moved to a different file. Where would it make sense?

File: src/useCases/common/SchemaRepository.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+const customFormats: Record<string, string> = {
+    "bkb-file": "FIL[A-Za-z0-9]{17}",
+    "bkb-relationship": "REL[A-Za-z0-9]{17}",
+    "bkb-relationshipRequest": "RRQ[A-Za-z0-9]{17}",
+    "bkb-message": "MSG[A-Za-z0-9]{17}",
+    "bkb-relationshipTemplate": "RLT[A-Za-z0-9]{17}",
+    "bkb-token": "TOK[A-Za-z0-9]{17}",
+    "bkb-relationshipChange": "RCH[A-Za-z0-9]{17}",
+    "bkb-device": "DVC[A-Za-z0-9]{17}"
+};
```
Comment (smelm) [inline src/useCases/common/SchemaRepository.ts:15]: These should probably be moved to a different file. Where would it make sense?

**Review by jkoenig134 — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -2,4 +2,4 @@ node_modules
 dist
 lib-web
 *.db
-*.log
+*.log
```
Comment (jkoenig134) [inline .gitignore:5]: newline

File: .prettierignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 dist
 lib-web
 node_modules
+src/useCases/common/Schemas.ts
```
Comment (jkoenig134) [inline .prettierignore:4]: newline

File: src/useCases/transport/files/DownloadFile.ts [hunk 5]
```diff
@@ -1,29 +1,32 @@
...
-    }
-}
-
 export interface DownloadFileResponse {
     content: Uint8Array;
     filename: string;
     mimetype: string;
 }
 
+class Validator extends SchemaValidator<DownloadFileRequest> {
```
Comment (jkoenig134) [inline src/useCases/transport/files/DownloadFile.ts:22]: Validator ist an der falschen stelle

Comment (jkoenig134) [reply src/useCases/transport/files/DownloadFile.ts:22]: (sollte schon einheitlich sein)

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**